packages/client/hmi-client/src/services/data.ts

```diff
@@ -49,6 +73,143 @@ const getXDDDictionaries = async () => {
 	return [] as XDDDictionary[];
 };
 
+const searchXDDDocuments = async (term: string, xddSearchParam?: XDDSearchParams) => {
+	const limitResultsCount = xddSearchParam?.perPage ?? XDD_RESULT_DEFAULT_PAGE_SIZE;
+
+	// NOTE when true it disables ranking of results
+	const enablePagination = xddSearchParam?.fullResults ?? false;
+
+	// "full_results": "Optional. When this parameter is included (no value required),
+	//  an overview of total number of matching documents is returned,
+	//  with a scan-and-scroll cursor that allows client to step through all results page-by-page.
+	//  NOTE: the "max" parameter will be ignored
+	//  NOTE: results may not be ranked in this mode
+	let searchParams = `term=${term}`;
+	const url = '/documents?';
+
+	if (xddSearchParam?.docid) {
```

Comment:
You should be able to use the `query-string` utility for this.
```typescript
import { toQueryString } from '@/utils/query-string';

const url = `/documents?${toQueryString(xddSearchParam)}`
```

Reply:
Yes, this part of the code needs to be corrected. I just moved functions around to remove `@ts-ignore` and `eslint-disable-line` that people used instead of organizing their code properly.